### DISCUSSION TITLE
How to import schema?

### DISCUSSION THREAD

**Discussion by mehemken** [kcl-lang/community#21] [Q&A]
Hello!
I'm very interested in using KCL at my company and trying to get comfortable with the project. I'm working on a proof of concept and I have a few questions. Here's my first question.

I am a FluxCD user, thus I have to configure [HelmReleases and HelmRepositories][1]. I'd like to use the upstream schemas for those. How do I import them?

The documentation clearly shows how to import Kubernetes schema:

```
❯ kcl mod add k8s:1.29
--- main.k:
import k8s.api.core.v1 as k8core
```

It is not obvious to me how to accomplish the same thing with other schema. For example I'd like to do this with Flux, Istio, Prometheus etc.

How do I do this? Can I help with documentation for it? (I am very new to contributing to open source)

[1]: https://fluxcd.io/flux/components/helm/helmreleases/ "helmrelease"

**Comment by mehemken** [comment]
Hmm, I've read a bit more and it seems like it has to be from an OCI registry?
There's no way to pull in schema from available go structs or OpenApiV3?

I'm comparing it to the capability of CUE lang. In CUE I can `go get ...` then `cue get go ...` and I have the schema available. Is there a way to do something similar in kcl?

**Comment by mehemken** [comment]
I've simply downloaded the CRD yaml from my cluster and used the instructions on the [kcl-openapi page][1].

However, this generates models for multiple versions of the same resource. For example, the flux git_repository has v1beta1 and v1beta2. When I run kcl this throws an error because of the duplicate naming. The solution to this is not obvious. I've resorted to manually deleting the older version. This is problematic because I may need both versions when doing an upgrade.

What is the recommended solution? Is it possible I missed a page in the documentation that already explains this?

[1]: https://github.com/kcl-lang/kcl-openapi "openapi"

**Reply by mehemken** [reply]
I've manually moved each version into its own directory. This gets rid of the errors. However this is unsustainable as the next time I auto-generate the models I will have to move the files again. Or I will need to write a script that does this.

Is there a way to tell kcl to do this for me?

**Comment by ozumun** [comment]
Ciao! È bello sapere che stai esplorando KCL per il tuo progetto. Per importare schemi come Flux, Istio o Prometheus, è necessario trovare o creare i moduli dello schema corrispondenti, poiché KCL non fornisce ancora il supporto integrato per tutti gli schemi di terze parti. È possibile definire autonomamente schemi personalizzati o cercare moduli forniti dalla comunità. Per Flux, potrebbe essere necessario definire manualmente gli schemi HelmRelease e HelmRepository in base ai loro CRD. Se sei interessato a contribuire, documentare questo processo o creare moduli riutilizzabili per questi strumenti sarebbe un modo fantastico per aiutare la comunità! Offerta [vlad-casino.it](https://vlad-casino.it/) di partenza ideale per nuovi giocatori.
